smart-contracts/contracts/mixins/MixinLockCore.sol

```diff
@@ -109,6 +125,19 @@ contract MixinLockCore is
     emit PriceChanged(oldKeyPrice, keyPrice);
   }
 
+  /**
+   * A function which lets the owner of the lock update the beneficiary account,
+   * which receives funds on withdrawal.
+   */
+  function updateBeneficiary(
+    address _beneficiary
+  ) external
+    onlyOwnerOrBeneficiary
```

Comment:
initially I thought this should be restricted to `onlyOwner`, but now I think this provides an interesting use case, and I can't think of a way a "bad actor" beneficiary could abuse this.